crates/ra_ide_api_light/src/assists/introduce_match.rs

```diff
@@ -0,0 +1,116 @@
+use ra_syntax::TextUnit;
+
+use crate::assists::{AssistCtx, Assist};
+
+pub fn introduce_match<'a>(ctx: AssistCtx) -> Option<Assist> {
+    let expr = ctx.covering_node();
+
+    ctx.build("introduce match", move |edit| {
+        let first_part = format!("match {} {{\n    ", expr.text());
+        let match_expr = format!("{}_ => (),\n}}", first_part);
+        edit.replace_node_and_indent(expr, match_expr.into_boxed_str());
+
+        // FIXME: Set cursor at beginning of first match arm.
+        // The following doesn't work because of re-indentation:
+        //
+        // edit.set_cursor(expr.range().start() + TextUnit::of_str(&first_part));
```

Comment:
Yeah... I think replace_and_indent could return an offset of the start of the substituted node?